gobblin-config-management/gobblin-config-core/src/main/java/org/apache/gobblin/config/store/zip/SimpleLocalIvyConfigStoreFactory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.config.store.zip;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.util.Properties;
+
+import org.apache.gobblin.config.store.api.ConfigStoreCreationException;
+import org.apache.hadoop.fs.Path;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+import com.google.common.collect.ImmutableMap;
+import com.sun.nio.zipfs.ZipFileSystem;
+
+
+/**
+ * Extension of {@link IvyConfigStoreFactory} that creates a {@link IvyConfigStoreFactory} which works for the
+ * local file system. Instead of downloading zip from jar artifactory, just use the testing zip files in classpath.
+ */
+public class SimpleLocalIvyConfigStoreFactory extends IvyConfigStoreFactory {
```

Comment:
Actually it is more than testing although the primary usage for that is testing.  It can also serve as using ZipConfigStore but have the back-up file sit in FS. One of the motivation for doing that is to instead of having a bunch of small files but only expose an `.zip` to FS.

Reply:
That's true. Only thing is that I don't think it makes much sense for this to extend `IvyConfigStoreFactory` since it's not even downloading through ivy and most code is not being shared.

Maybe instead this class could be named `ZipFileConfigStoreFactory`, and `IvyConfigStoreFactory` extends it? Then have a separate method for getting the zip FS so that Ivy one can override it and do the download there and rest of the code can be shared.

Reply:
I see, got your point, that makes sense. Will address.